prettier, linting, fixed tests

nerem opened the pull request.

Commit:
prettier, linting, fixed tests

Commit:
added github action for frontend

xerus-github commented:
When the FE is started, there are a couple of warnings, such as: 

WARNING in ./node_modules/@hqoss/monads/dist/lib/either/either.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '<...>/frontend/node_modules/@hqoss/monads/lib/either/either.ts' file: Error: ENOENT: no such file or directory, open '<...>/frontend/node_modules/@hqoss/monads/lib/either/either.ts'

Are some source maps missing after the update? Do we need them?

xerus-github commented on frontend/src/components/App/App.test.tsx:
```diff
@@ -13,75 +13,75 @@ const mockedGetFeed = getFeed as jest.Mock<ReturnType<typeof getFeed>>;
-  await act(async () => {
-    await render(<App />);
-  });
+        render(<App />);
```
Also 
` 
        await act(async () => {
            render(<App />);
        });
`  ? There is also a "warning"/error in test output. The other deprecated warnings are ok, of course "The object notation for `createSlice.extraReducers` is deprecated, and will be removed in RTK 2.0."

nerem replied:
done

xerus-github commented:
LGTM. (Rebased the other BE stuff temporarily onto this branch and everything still works)

xerus-github commented:
Can you run another "lint:fix" on your branch? --> `ArcticlePage.test.txt`

Commit:
- fixed warnings in a test by using await act.
- fixed warning when starting the app by suppressing the warnings. Now you can also see the code of the application in the browser dev tools, which hasn't been the case before.
- fixed some missing linting

nerem commented:
> Can you run another "lint:fix" on your branch? --> `ArcticlePage.test.txt`

done

nerem commented:
> When the FE is started, there are a couple of warnings, such as:
> 
> WARNING in ./node_modules/@hqoss/monads/dist/lib/either/either.js Module Warning (from ./node_modules/source-map-loader/dist/cjs.js): Failed to parse source map from '<...>/frontend/node_modules/@hqoss/monads/lib/either/either.ts' file: Error: ENOENT: no such file or directory, open '<...>/frontend/node_modules/@hqoss/monads/lib/either/either.ts'
> 
> Are some source maps missing after the update? Do we need them?

done
